cg/constants.py

```diff
@@ -67,64 +67,70 @@
 # presence of mandatory files. Keys = tags found in deliverables, values = MIP standard tags and
 # mandatory flag
 MIP_DNA_TAGS = {
+    tuple(["config_analysis"]): {"tags": ["mip-config"], "is_mandatory": True},
```

Comment:
Great, one small remark. Why do we create tuples with `tuple` all the time?
In python:
```python
tuple(['a','b']) == ('a','b')
```
BUT remember that
```python
('a','b') != ('b','a')
```
which also means that
```python
tuple(['a','b']) != tuple(['b','a'])
```

Reply:
```python
Python 3.7.7 (default, Apr 21 2020, 11:51:15)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.13.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: assert tuple(['a','b']) == ('a','b')

In [2]: assert ('a','b') != ('b','a')

In [3]: assert tuple(['a','b']) != tuple(['b','a'])

In [4]: assert tuple(['a','b']) == tuple(['b','a'])
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-4-5ace5e5729c5> in <module>
----> 1 assert tuple(['a','b']) == tuple(['b','a'])

AssertionError:
```

Reply:
Dictionary keys must be hashable, so I went with tuples.

![image](https://user-images.githubusercontent.com/26384108/84982901-e9ad7c80-b137-11ea-9397-6d3e6ae6a39a.png)

I also check if tags are *in* the keys, not if the tags *are* the keys.

Do see your point though. I realise it works fine for my code, but it's a bit tricky for other developers that don't know the reasoning behind it.

Reply:
If you want I can remove the tuple constructor and just let it be a tuple by just keeping the `()`.

Reply:
Defining singleton tuples like that will make them of the type of the object in them, you would need a trailing comma to force the type tuple:

![image](https://user-images.githubusercontent.com/26384108/84992257-f508a400-b147-11ea-9dbc-e9d1c3ab63b5.png)

So it would need to be 
`("config_analysis",): {"tags": ["mip-config"], "is_mandatory": True},`
or since we can check for more than one distinct value in the key
`(["config_analysis"],): {"tags": ["mip-config"], "is_mandatory": True},` and 
`(["multiqc_ar", "html"],): {"tags": ["multiqc-html"], "is_mandatory": True},` and so on.

I didn't like that so I choose to use `tuple(...)`, because it looked more clear to me.

I'm not opposed to your suggestion, I'm just explaining the reasoning behind my choice :)

Reply:
Thx @barrystokman ! That made it more clear. I removed the tuple and added a trailing comma. I think this will be great - it is a little less clutter and still very clear that this is a tuple. I hope that is fine!

Reply:
Ok I see your point @barrystokman , I'm in favour of

```python
("config_analysis",): {"tags": ["mip-config"], "is_mandatory": True}
```
But I won't object if you declare by using the word `tuple`

Reply:
If you check if the tag is IN the tuple I would choose to not use tuples at all and just have unique strings as keys and duplicate values instead, example:

now
```python
{tuple(["config_analysis", "config"]): {"tags": ["mip-config"], "is_mandatory": True}}
```
could be
```python
{
    "config_analysis": {"tags": ["mip-config"], "is_mandatory": True},
    "config": {"tags": ["mip-config"], "is_mandatory": True}
}
```
This will make the dictionary search more effective, at least if I understand what you mean. I thought we used tuples as keys because we search the dictionary with tuples...

Reply:
```
("multiqc_ar", "html"): {"tags": ["multiqc-html"], "is_mandatory": True},
("multiqc_ar", "json"): {"tags": ["multiqc-json"], "is_mandatory": True},
```

Here's a case where the outcome is dependent on two tags in the deliverables file, these can't be mapped one-on-one. I basically check per file in the deliverables.yaml if all the tags in the key (for instance `multiqc_ar` and `html`) are in the tags of the file (the number of files can be anything, so like `some_tag`, `another_tag`, `html`, `tag4: the taggening`, `multiqc_ar`. If that's the case I know the Housekeeper tag should be `multiqc-html` and not the other one.

I'm pretty sure more experienced developers can come up with a much more elegant solution, but it's the best I could come up with :frowning_face:

Reply:
Yes that sounds a bit tricky... so I guess you have to loop over all tags in the dictionary every time you want to check what tags to fetch? Where is the code for doing that?